src/Command/PostsMigrator.php

```diff
@@ -772,6 +832,244 @@ public function cmd_hide_featured_images( $positional_args, $assoc_args ) {
+			// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery
+			$update_result = $wpdb->update(
+				$wpdb->posts,
+				$update_data,
+				[
+					'ID' => $post_id,
+				]
+			);
+
+			if ( ! $update_result ) {
+				$logger->error( 'Couldn\'t update post.' );
+
+				continue;
+			}
+
+			clean_post_cache( $post_id );
+
+			$logger->info( '✅ Successfully updated post' );
+
+			if ( ! empty( $category_id ) ) {
+				$category_updated = wp_set_post_categories( $post_id, $category_id, true );
+
+				if ( false === $category_updated || is_wp_error( $category_updated ) ) {
+					$logger->error( sprintf( '🚫 Couldn\'t set category "%s" for post', $category_name ) );
+				} else {
+					$logger->info( sprintf( '✅ Successfully set category "%s" for post', $category_name ) );
+				}
+			}
+
+			/**
+			 * Filters the source URL of the post being migrated.
+			 * The Source URL should contain the relative URL of the Post before the migration.
+			 * 
+			 * @param string $source_url The source URL of the post in the format /{POST_TYPE}/{POST_NAME}.
+			 * @param int    $post_id    The post ID.
+			 * @param string $post_name  The original post name before migration.
+			 */
+			$source_url = apply_filters(
+				'nmt_posts_migrator_cpt_to_posts_post_source_url',
+				sprintf( '/%s/%s', $post_type, $post_name ),
+				$post_id,
+				$post_name
+			);
+
+			/**
+			 * Filters the target URL of the post being migrated.
+			 * The Target URL should contain the relative URL of the Post after the migration.
+			 * 
+			 * @param string $target_url The target URL of the post in the format /{POST_NAME}.
+			 * @param int    $post_id    The post ID.
+			 * @param string $post_name  The updated post name after migration.
+			 */
+			$target_url = apply_filters(
+				'nmt_posts_migrator_cpt_to_posts_post_target_url',
+				str_replace( home_url(), '', get_permalink( $post_id ) ),
+				$post_id,
+				$new_post_name
+			);
+
+			// Populate CSV redirection record.
+			$redirects_csv_writer->put( [ $source_url, $target_url ] );
+
+			/**
+			 * Fires once the Post has been migrated.
+			 * 
+			 * @param int $post_id Post ID.
+			 */
+			do_action( 'nmt_posts_migrator_cpt_to_posts_post_migrated', $post_id );
+		}
+
+		$redirects_csv_writer->close();
+		$progress_bar->finish();
+
+		$logger->info( '🏁 Migration completed successfully!' );
+
+		wp_cache_flush();
+	}
+
+	/**
+	 * Callable for `newspack-content-migrator rollback-posts-to-cpt`.
+	 *
+	 * @param array $pos_args   Positional arguments.
+	 * @param array $assoc_args Associative arguments.
+	 * @return void
+	 */
+	public function cmd_rollback_posts_to_cpt( array $pos_args, array $assoc_args ): void {
+		$post_type     = $assoc_args['post_type'] ?? null;
+		$category_name = $assoc_args['category_name'] ?? null;
+
+		if ( ! $post_type ) {
+			WP_CLI::error( 'Post Type is required' );
+
+			return;
+		}
+
+		$category = get_term_by( 'name', $category_name, 'category' );
```

Comment:
I know because I've dealt with this, but if you have a category name like `Arts & Craft`, `get_term_by` will not return the category correctly. So I recommend using slug instead (where you `sanititze_title($category_name)`) or you use a direct DB query to get the category.